### ISSUE TITLE
If translation is missing fallback to english or something else

### ISSUE THREAD

**Issue by adirkuhn** [myparcelnl/woocommerce#1177]
### Version check

- [X] My version of the plugin is v5.0.0-beta.0 or above.

### WordPress version

6.6.2

### PHP version

8.1

### What went wrong?

When no translation file is found: kaboom

I would suggest to fallback to english or dutch

### Reproduction steps

1. Install the plugin in a non dutch, english or french site

### Relevant log output

```shell
[Thu Oct 10 21:54:32.588991 2024] [php:error] [pid 847365] [client 64.252.79.142:0] PHP Fatal error:  Uncaught InvalidArgumentException: File "/var/www/html/wp-content/plugins/woocommerce-myparcel//config/pdk/translations/pt.json" does not exist. in /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Base/FileSystem.php:46\nStack trace:\n#0 /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Language/Service/AbstractLanguageService.php(54): MyParcelNL\\Pdk\\Base\\FileSystem->get()\n#1 /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Base/Repository/Repository.php(52): MyParcelNL\\Pdk\\Language\\Service\\AbstractLanguageService->MyParcelNL\\Pdk\\Language\\Service\\{closure}()\n#2 /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Language/Repository/LanguageRepository.php(17): MyParcelNL\\Pdk\\Base\\Repository\\Repository->retrieve()\n#3 /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Language/Service/AbstractLanguageService.php(55): MyParcelNL\\Pdk\\Language\\Repository\\LanguageRepository->getTranslations()\n#4 /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Language/Service/AbstractLanguageService.php(75): MyParcelNL\\Pdk\\Language\\Service\\AbstractLanguageService->getTranslations()\n#5 /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Base/Facade.php(25): MyParcelNL\\Pdk\\Language\\Service\\AbstractLanguageService->translate()\n#6 /var/www/html/wp-content/plugins/woocommerce-myparcel/src/Hooks/AbstractFieldsHooks.php(22): MyParcelNL\\Pdk\\Base\\Facade::__callStatic()\n#7 /var/www/html/wp-content/plugins/woocommerce-myparcel/src/Hooks/SeparateAddressFieldsHooks.php(84): MyParcelNL\\WooCommerce\\Hooks\\AbstractFieldsHooks->createField()\n#8 /var/www/html/wp-content/plugins/woocommerce-myparcel/src/Hooks/SeparateAddressFieldsHooks.php(30): MyParcelNL\\WooCommerce\\Hooks\\SeparateAddressFieldsHooks->extendWithSeparateAddressFields()\n#9 /var/www/html/wp-includes/class-wp-hook.php(324): MyParcelNL\\WooCommerce\\Hooks\\SeparateAddressFieldsHooks->extendBillingFields()\n#10 /var/www/html/wp-includes/plugin.php(205): WP_Hook->apply_filters()\n#11 /var/www/html/wp-content/plugins/woocommerce/includes/class-wc-countries.php(1727): apply_filters()\n#12 /var/www/html/wp-content/plugins/woocommerce/includes/class-wc-checkout.php(253): WC_Countries->get_address_fields()\n#13 /var/www/html/wp-content/plugins/woo-stripe-payment/includes/wc-stripe-functions.php(1290): WC_Checkout->get_checkout_fields()\n#14 /var/www/html/wp-content/plugins/woo-stripe-payment/includes/class-wc-stripe-frontend-scripts.php(129): wc_stripe_get_checkout_fields()\n#15 /var/www/html/wp-includes/class-wp-hook.php(324): WC_Stripe_Frontend_Scripts->localize_scripts()\n#16 /var/www/html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()\n#17 /var/www/html/wp-includes/plugin.php(517): WP_Hook->do_action()\n#18 /var/www/html/wp-includes/script-loader.php(2215): do_action()\n#19 /var/www/html/wp-includes/class-wp-hook.php(324): wp_print_head_scripts()\n#20 /var/www/html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()\n#21 /var/www/html/wp-includes/plugin.php(517): WP_Hook->do_action()\n#22 /var/www/html/wp-includes/general-template.php(3065): do_action()\n#23 /var/www/html/wp-content/themes/bacola/header.php(17): wp_head()\n#24 /var/www/html/wp-includes/template.php(810): require_once('...')\n#25 /var/www/html/wp-includes/template.php(745): load_template()\n#26 /var/www/html/wp-includes/general-template.php(48): locate_template()\n#27 /var/www/html/wp-content/themes/bacola/404.php(10): get_header()\n#28 /var/www/html/wp-includes/template-loader.php(106): include('...')\n#29 /var/www/html/wp-blog-header.php(19): require_once('...')\n#30 /var/www/html/index.php(17): require('...')\n#31 {main}\n  thrown in /var/www/html/wp-content/plugins/woocommerce-myparcel/vendor/myparcelnl/pdk/src/Base/FileSystem.php on line 46, referer: https://cantinhobrasileiro.nl/
```

**Comment by EdieLemoine** [comment]
Hi @adirkuhn, thanks for reporting. This was intended to be the functionality but it turns out there was a small bug in the implementation. It'll be fixed in the next release. 😄

**Comment by EdieLemoine** [comment]
The bugfix was not merged before the v5.0.0 release, but you can probably expect it in the next one.
